lib/internal/modules/package_json_reader.js

```diff
@@ -37,36 +55,40 @@ function deserializePackageJSON(path, contents) {
   } = contents;
 
   // This is required to be used in getPackageScopeConfig.
-  if (optionalFilePath) {
-    pjsonPath = optionalFilePath;
-  }
-
-  // The imports and exports fields can be either undefined or a string.
-  // - If it's a string, it's either plain string or a stringified JSON string.
-  // - If it's a stringified JSON string, it starts with either '[' or '{'.
-  const requiresJSONParse = (value) => (value !== undefined && (value[0] === '[' || value[0] === '{'));
+  const pjsonPath = optionalFilePath ?? path;
 
   return {
-    __proto__: null,
-    exists: true,
-    pjsonPath,
-    name,
-    main,
-    type,
-    // This getters are used to lazily parse the imports and exports fields.
-    get imports() {
-      const value = requiresJSONParse(plainImports) ? JSONParse(plainImports) : plainImports;
-      ObjectDefineProperty(this, 'imports', { __proto__: null, value });
-      return this.imports;
-    },
-    get exports() {
-      const value = requiresJSONParse(plainExports) ? JSONParse(plainExports) : plainExports;
-      ObjectDefineProperty(this, 'exports', { __proto__: null, value });
-      return this.exports;
+    data: {
+      __proto__: null,
+      ...(name != null && { __proto__: null, name }),
+      ...(main != null && { __proto__: null, main }),
+      ...(type != null && { __proto__: null, type }),
```

Comment:
null prototype is not useful here, spread only copies own properties
```suggestion
      ...(name != null && { name }),
      ...(main != null && { main }),
      ...(type != null && { type }),
```